addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -776,7 +776,6 @@ def _prepare_move_vals(self):
 
         if self.payment_mode == 'company_account':
             to_return['date'] = most_recent_expense or today
-            to_return['invoice_date'] = today
```

Comment:
I think this should remains:
`_prepare_move_vals` is called in `_prepare_bills_vals` just above and, as the name suggest: `'move_type': 'in_invoice',`

Reply:
Yes but no...
The architecture of the module sucks and `_prepare_bills_vals` is only called for `company_account`.
The logic is just dispersed all over the place.

Reply:
We can do something a bit crappy but that would be fixing the situation for the payment:
In `addons/hr_expense/models/hr_expense.py` in method: `def _prepare_payments_vals(self):`
Just add an `'invoice_date': False` after the call or pop the value from the returned dict before creating the final dict.

It won't be nice but that would allow keeping the `invoice_date` for the bill move.
Maybe someone like @JulienAlardot will clean it later on?

Reply:
But we keep the invoice date for the bills.
We never enter this branch of `if self.payment_mode == 'company_account':` when it is a bill

We only call `_prepare_bills_vals` when `payment_mode == 'own_account'`
```python
own_account_sheets = self.filtered(lambda sheet: sheet.payment_mode == 'own_account')
...
moves = self.env['account.move'].create([sheet._prepare_bills_vals() for sheet in own_account_sheets])
```

Reply:
`_prepare_bills_vals` is only called for `own_account`, but IMO we can remove that line
I don't know why this is there in the first place, don't remember seeing it when I looked at the PR introducing it, but I may have missed it. 
[This if](https://github.com/odoo/odoo/pull/163511/files/7192d68b8e91ba8f5d8e9df5b81f6855f025d16e#diff-42c75e4d29952aadf0abdd1bfe7e066bef3db1275c16d64dd3f3b96a332a2ff3L777-L780) should have been done respectively in the _prepare_bills and _prepare_payments as it complicates everything. :thinking: